Reqnroll.Tools.MsBuild.Generation/GenerateFeatureFileCodeBehindTask.cs

```diff
@@ -1,69 +1,76 @@
 {
-    public class GenerateFeatureFileCodeBehindTask : Task
-    {
-        public IFeatureFileCodeBehindGenerator CodeBehindGenerator { get; set; }
-        public IAnalyticsTransmitter AnalyticsTransmitter { get; set; }
+    public const string CodeBehindFileMetadata = "CodeBehindFile"; //in
+    public const string MessagesFileMetadata = "MessagesFile"; //in,out
+    public const string MessagesResourceNameMetadata = "MessagesResourceName"; //out
 
-        [Required]
-        public string ProjectPath { get; set; }
+    public IGenerateFeatureFileCodeBehindTaskDependencyCustomizations DependencyCustomizations { get; set; }
 
-        public string RootNamespace { get; set; }
+    [Required]
+    public string ProjectPath { get; set; }
 
-        public string ProjectFolder => Path.GetDirectoryName(ProjectPath);
-        public string OutputPath { get; set; }
-        public string IntermediateOutputPath { get; set; }
-        public ITaskItem[] FeatureFiles { get; set; }
+    public string RootNamespace { get; set; }
 
-        public ITaskItem[] GeneratorPlugins { get; set; }
+    public string ProjectFolder => Path.GetDirectoryName(ProjectPath);
+    public string OutputPath { get; set; }
+    public ITaskItem[] FeatureFiles { get; set; }
 
-        [Output]
-        public ITaskItem[] GeneratedFiles { get; private set; }
+    public ITaskItem[] GeneratorPlugins { get; set; }
 
-        public string MSBuildVersion { get; set; }
-        public string AssemblyName { get; set; }
-        public string TargetFrameworks { get; set; }
-        public string TargetFramework { get; set; }
-        public string ProjectGuid { get; set; }
+    [Output]
+    public ITaskItem[] GeneratedFiles { get; private set; }
 
-        public override bool Execute()
-        {
-            var generateFeatureFileCodeBehindTaskContainerBuilder = new GenerateFeatureFileCodeBehindTaskContainerBuilder();
-            var generatorPlugins = GeneratorPlugins?.Select(gp => gp.ItemSpec).Select(p => new GeneratorPluginInfo(p)).ToArray() ?? Array.Empty<GeneratorPluginInfo>();
-            var featureFiles = FeatureFiles?.Select(i => i.ItemSpec).ToArray() ?? Array.Empty<string>();
+    public string MSBuildVersion { get; set; }
+    public string AssemblyName { get; set; }
+    public string TargetFrameworks { get; set; }
+    public string TargetFramework { get; set; }
+    public string ProjectGuid { get; set; }
 
-            var msbuildInformationProvider = new MSBuildInformationProvider(MSBuildVersion);
-            var generateFeatureFileCodeBehindTaskConfiguration = new GenerateFeatureFileCodeBehindTaskConfiguration(AnalyticsTransmitter, CodeBehindGenerator);
-            var generateFeatureFileCodeBehindTaskInfo = new ReqnrollProjectInfo(generatorPlugins, featureFiles, ProjectPath, ProjectFolder, ProjectGuid, AssemblyName, OutputPath, IntermediateOutputPath, RootNamespace, TargetFrameworks, TargetFramework);
+    public bool DebugMSBuildTask { get; set; }
```

Comment:
Where tasks in other project have a property to launch the debugger, they call it `LaunchDebugger` - we might want to follow that convention.

Reply:
Makes sense. I have just found this prop unused in our codebase but didn't think much of the name.

Reply:
Changed.